lib/internal/fs/streams.js

```diff
@@ -120,29 +120,29 @@ function close(stream, err, cb) {
 }
 
 function importFd(stream, options) {
-  stream.fd = null;
-  if (options.fd != null) {
-    if (typeof options.fd === 'number') {
-      // When fd is a raw descriptor, we must keep our fingers crossed
-      // that the descriptor won't get closed, or worse, replaced with
-      // another one
-      // https://github.com/nodejs/node/issues/35862
-      stream.fd = options.fd;
-    } else if (typeof options.fd === 'object' &&
-      options.fd instanceof FileHandle) {
-      // When fd is a FileHandle we can listen for 'close' events
-      if (options.fs)
-        // FileHandle is not supported with custom fs operations
-        throw new ERR_METHOD_NOT_IMPLEMENTED('FileHandle with fs');
-      stream[kHandle] = options.fd;
-      stream.fd = options.fd.fd;
-      stream[kFs] = FileHandleOperations(stream[kHandle]);
-      stream[kHandle][kRef]();
-      options.fd.on('close', FunctionPrototypeBind(stream.close, stream));
-    } else
-      throw ERR_INVALID_ARG_TYPE('options.fd',
-                                 ['number', 'FileHandle'], options.fd);
+  if (typeof options.fd === 'number') {
+    // When fd is a raw descriptor, we must keep our fingers crossed
+    // that the descriptor won't get closed, or worse, replaced with
+    // another one
+    // https://github.com/nodejs/node/issues/35862
+    stream[kFs] = options.fs || fs;
+    return options.fd;
+  } else if (typeof options.fd === 'object' &&
+             options.fd instanceof FileHandle) {
+    // When fd is a FileHandle we can listen for 'close' events
+    if (options.fs) {
+      // FileHandle is not supported with custom fs operations
+      throw new ERR_METHOD_NOT_IMPLEMENTED('FileHandle with fs');
```

Comment:
The error message here could be more descriptive (could just use the same message as the comment)

Reply:
The error message currently is `The FileHandle with fs method is not implemented`. `ERR_METHOD_NOT_IMPLEMENTED` may not be the fittest here, but I'm afraid changing it would be a breaking change, best left to another PR. Or do you have a suggestion that would work with `ERR_METHOD_NOT_IMPLEMENTED`?